File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s378126458_commented.py
# Reading input
import sys
input = sys.stdin.read
data = input().split()

N = int(data[0])
Q = int(data[1])
s = data[2]

# Initialize prefix sum array
pre = [0] * (N + 1)

# Building the prefix sum array
for i in range(1, N):
    if s[i-1] == 'A' and s[i] == 'C':
        pre[i] = 1
    pre[i] += pre[i-1]

# Querying
for _ in range(Q):
    l, r = int(data[3 + 2 * _]), int(data[4 + 2 * _])
    r -= 1
    print(pre[r] - pre[l-1])

# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C/s378126458_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    unsigned int N, Q;
    scanf("%u %u", &N, &Q);
    char s[N+1];
    scanf("%s", s);
    int pre[N+1];
    memset(pre, 0, sizeof(pre));
    
    for(size_t i = 1; i < N; i++) {
        if(s[i-1] == 'A' && s[i] == 'C') {
            pre[i] = 1; // initially the prefix sum array is the k array
        }
    }
    
    // Building the psa
    for(size_t i = 1; i < N; i++) {
        pre[i] += pre[i-1];
    }
    
    // Querying
    while(Q-- > 0) {
        unsigned int l, r;
        scanf("%u %u", &l, &r);
        r--;
        printf("%d\n", pre[r] - pre[l-1]);
    }
    
    // End of Code
}
